### PR TITLE
Add Improved Fog and Lighting

### PR DESCRIPTION
**Pull request by ikonomov**
lightingEnhancerGroup allows the correct load order

### CODE REVIEW

File: masterlist.yaml [hunk 1]
```diff
@@ -3544,7 +3544,13 @@ plugins:
...
     tag: [ Graphics ]
 
-  - name: 'ImprovedHearthfireLighting.esp'
+- name: 'ImprovedFog&Lighting.esp'
+  url:
+   - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/46289/'
+     name: 'Improved Fog and Lighting'
+  group: *lightingEnhancerGroup
+
+- name: 'ImprovedHearthfireLighting.esp'
```
Comment (sibir-ine) [inline masterlist.yaml]: The indentation needs to be corrected:

```suggestion
  - name: 'ImprovedFog&Lighting.esp'
    url:
      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/46289/'
        name: 'Improved Fog and Lighting'
    group: *lightingEnhancerGroup

  - name: 'ImprovedHearthfireLighting.esp'
```

**Review by sibir-ine — CHANGES_REQUESTED** [review]
@ikonomov, the group placement is fine. However, it'd be a good idea to mark any other plugins in the same group as incompatible or add load rules for them (though, from a cursory look, they all appear incompatible) to avoid cyclic interactions. Here's a list of them:

- [`ELE_SSE.esp`](https://www.nexusmods.com/skyrimspecialedition/mods/1377)
- [`ELFXEnhancer.esp`](https://www.nexusmods.com/skyrimspecialedition/mods/2424)
- [`ELFX - Hardcore.esp`](https://www.nexusmods.com/skyrimspecialedition/mods/2424)
- [`Luminosity Lighting Overhaul.esp`](https://www.nexusmods.com/skyrimspecialedition/mods/16830)
- [`WSEnhancer.esp`](https://www.nexusmods.com/skyrimspecialedition/mods/37831)

**Comment by ikonomov** [comment]
@sibir-ine thank you for the edits and suggestions.  The mod edits both interior lighting and weather records, I think you are right, most of the other mods in this group are probably incompatible.  Weather mods should be OK though, but there could be conflicts with interior mods so at least the popular interior mods should be specified as incompatible.  I've made all your suggested changes.

File: masterlist.yaml [hunk 2]
```diff
@@ -3544,6 +3544,22 @@ plugins:
         condition: 'not (file("Particle Patch for ENB.bsa") and active("Particle Patch for ENB.esp")) and not file("textures/effects/gradients/gradillusiondark01m.dds") and not file("textures/WiZkiD/iron_e.dds")'
     tag: [ Graphics ]
 
+  - name: 'ImprovedFog&Lighting.esp'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/46289/'
+        name: 'Improved Fog and Lighting'
+    group: *lightingEnhancerGroup
+    inc:
+      - 'ELE_SSE.esp'
```
Comment (sibir-ine) [inline masterlist.yaml]: There's some extra whitespace here:

```suggestion
      - 'ELE_SSE.esp'
```

File: masterlist.yaml [hunk 3]
```diff
@@ -3544,6 +3544,22 @@ plugins:
...
+  - name: 'ImprovedFog&Lighting.esp'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/46289/'
+        name: 'Improved Fog and Lighting'
+    group: *lightingEnhancerGroup
+    inc:
+      - 'ELE_SSE.esp'    
+      - 'ELFXEnhancer.esp'
+      - 'ELFX - Hardcore.esp'
+      - 'EnhancedLightsandFX.esp'
```
Comment (sibir-ine) [inline masterlist.yaml]: Both plugins have `EnhancedLightsandFX.esp` as a master, so listing it as incompatible is sufficient:

```suggestion
      - 'EnhancedLightsandFX.esp'
```

File: masterlist.yaml [hunk 4]
```diff
@@ -3544,6 +3544,22 @@ plugins:
...
+      - 'Luminosity Lighting Overhaul.esp'
+      - 'Luminosity - Lightweight Lighting.esp'
```
Comment (sibir-ine) [inline masterlist.yaml]: This version of LLO is probably outdated enough to not warrant worrying about it:

```suggestion
```

**Review by sibir-ine — CHANGES_REQUESTED** [review]
Just needs a bit more cleanup & it'll be good to go.

**Comment by ikonomov** [comment]
> Just needs a bit more cleanup & it'll be good to go.

xD every commit of mine has to be messy, it seems.  Thank you once again!

**Review by sibir-ine — APPROVED** [review]
No worries, thanks for the contribution!

**Pull request merged**